api/python/tests/test_cli.py

```diff
@@ -41,5 +42,38 @@ def test_push(self):
                 main.main(('push', '--dir', tmp_dir, name))
 
                 mocked_package_class.assert_called_once_with(quilt3.Package)
-                mocked_set_dir.assert_called_once_with('.', tmp_dir)
+                mocked_set_dir.assert_called_once_with('.', tmp_dir, meta=None)
                 mocked_push.assert_called_once_with(name, registry=None, dest=None, message=None)
+
+
+@pytest.mark.parametrize(
+    'arg, meta, expected_set_dir_count, expected_push_count, expected_meta',
+    [
+        (None, None, 1, 1, None),
+        ('--meta', '{invalid: meta}', 0, 0, {}),
+        ('--meta', "{'single': 'quotation'}", 0, 0, {}),
+        ('--meta', '{"test": "meta", }', 0, 0, {}),
+        ('--meta', '{"test": "meta"}', 1, 1, {"test": "meta"}),
+    ]
+)
+def test_push_with_meta_data(arg, meta, expected_set_dir_count, expected_push_count, expected_meta):
+    name = 'test/name'
+    pkg = quilt3.Package()
+
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        (Path(tmp_dir) / 'foo').touch()
+        (Path(tmp_dir) / 'bar').mkdir()
+        (Path(tmp_dir) / 'bar' / 'baz')
+
+        with mock.patch('quilt3.Package.__new__', return_value=pkg) as mocked_package_class,\
+             mock.patch.object(pkg, 'set_dir', wraps=pkg.set_dir) as mocked_set_dir, \
+             mock.patch.object(pkg, 'push') as mocked_push:
+
+            if arg:
+                main.main(('push', '--dir', tmp_dir, name, arg, meta))
```

Comment:
where does --regsitry come from? fine if there's a default but we should note the same

Reply:
I don't understand the question though. This test is for --meta data and not --registry and also registry is optional.

Reply:
this is ok as is but you can note in a comment that registry is optional